src/cursor/cur_table.c

```diff
@@ -623,6 +623,29 @@ err:	CURSOR_UPDATE_API_END(session, ret);
 }
 
 /*
+ * __curtable_reserve --
+ *	WT_CURSOR->reserve method for the table cursor type.
+ */
+static int
+__curtable_reserve(WT_CURSOR *cursor)
+{
+	WT_CURSOR_TABLE *ctable;
+	WT_DECL_RET;
+	WT_SESSION_IMPL *session;
+
+	ctable = (WT_CURSOR_TABLE *)cursor;
+	JOINABLE_CURSOR_UPDATE_API_CALL(cursor, session, update, NULL);
+	WT_ERR(__curtable_open_indices(ctable));
```

Comment:
> I don't understand why this is necessary, since reserve isn't looking at indices.

I went back-and-forth on this one, but I think it's reasonable to leave it in: it makes the reserve method's initial cursor code look the same as the other methods, it's not a performance cost because the eventual update will call `__curtable_open_indices` anyway, and there's general error checking in `__curtable_open_indices` (for example, checking for bulk cursors). The reserve method is an update under the covers, and I'd rather the indices be open before we do updates.